### ISSUE TITLE
[Metrics/***Length] Is it possible to exclude heardocs line count?

### ISSUE THREAD

**Issue by KosukeOhmura** [rubocop-hq/rubocop#5394]
## Expected behavior

Hi,
Metrics/ClassLength, Metrics/MethodLength and other cops for line count doesn't ignore heardocs.
I can ignore it by writing `# rubocop:disable Metrics/MethodLength` for all points that have heardocs, but it is not a smart way. Can I configure to ignore heardoc's line count?

## Actual behavior

When I write this code:

```
def self.list(gender)
  sql = <<-SQL
    SELECT
      uuid
      ,name
      ,tel
      ,email
      ,gender
      ,created_at
      ,updated_at
    FROM
      get_member_list(p_gender := :gender)
  SQL

  find_by_sql([sql, { gender: gender }])
end
```

I get error `C: Metrics/MethodLength: Method has too many lines.`.

## Steps to reproduce the problem

Write a method that have a long heardoc and then execute `rubocop`.

my .rubocop.yml is this:

```
AllCops:
  TargetRubyVersion: 2.5
AsciiComments:
  Enabled: false
Metrics/LineLength:
  Max: 100
  IgnoredPatterns: ['(\A|\s)#']

Style/EmptyMethod:
  Enabled: false
Style/ClassAndModuleChildren:
  EnforcedStyle: compact
  AutoCorrect: true
```

## RuboCop version

```
$ rubocop -V
0.52.1 (using Parser 2.4.0.2, running on ruby 2.5.0 x86_64-linux)
```

Thanks

**Comment by bbatsov** [comment]
I'm closing this ticket due to no recent activity. Feel free to re-open it if you ever come back to it.

**Comment by imdrasil** [comment]
Hi there. I came across almost same question as described here. But my question is slightly wider: how to satisfy rubocop with method that represents one unique (this means that it is almost useless to move it to a separate class) query with heavy SQL usage. Here is an example:

```ruby
def request
  Sales
    .select(
      <<~SQL
        sales.id,
        sales.business_id,
        sales.item_id,
      SQL
    )
    .joins(:business)
    .joins(
      <<~SQL
        INNER JOIN fiscal_periods current_fiscal_period
          ON current_fiscal_period.id = sales.fiscal_period_id
        LEFT OUTER JOIN sales actuals
           ON  actuals.business_id      = sales.business_id
           AND actuals.item_id          = sales.item_id
           AND actuals.fiscal_period_id = fiscal_periods.id
      SQL
    )
    .where(sales: { fiscal_period_id: fiscal_period.id })
    .group(:id, :business_id, :item_id)
end
```

This is reduced example. I totally agree wit `Metrics/MethodLength` that this is too much lines for one method. But how to handle such a case? Move custom SQL to a separate methods - this will reduce readability a lot (e.g. aliases defined in some places are used in others). Making separate class per each such request also not always a pretty good idea. 

Does anybody have any idea regarding this?

**Comment by schmijos** [comment]
As of https://github.com/rubocop-hq/rubocop/pull/8159 you can configure heredocs to be counted as one line (`CountAsOne`).
